ivy/functional/frontends/jax/devicearray.py

```diff
@@ -73,6 +73,16 @@ def mean(self, *, axis=None, dtype=None, out=None, keepdims=False, where=None):
             keepdims=keepdims,
             where=where,
         )
+    
+    def cumprod(self, axis=None, dtype=None, out=None):
+        if dtype is None:
+            dtype = ivy.as_ivy_dtype(self.dtype)
+        return jax_frontend.numpy.cumsum(
```

Comment:
Did you use `cumsum` instead of `cumprod` by mistake here?

Reply:
As for the dtype error, I think you should first check to see if the frontend function you are using passes its test.

Reply:
Ohhh sorry, as I have sent PR for `cumsum` and `cumprod` both, that somehow got mixed. And I think that was the reason I was receiving the error.